model-optimizer/mo/utils/ir_engine/ir_engine.py

```diff
@@ -404,3 +409,70 @@ def __eq__(self, other):
         if not isinstance(other, IREngine):
             raise AttributeError("IREngine can be compared only with IREngine object type")
         return self.compare(other)[0]
+
+    def __read_if(self, layer, layer_attrs):
+        xml_then_body_child = list(layer.iterfind('then_body'))
+        xml_else_body_child = list(layer.iterfind('else_body'))
+        assert len(xml_then_body_child) == 1 or len(xml_else_body_child) == 1
+
+        then_body_ir = IREngine(path_to_xml=None,
+                                path_to_bin=self.path_to_bin,
+                                xml_tree=ElementTree(xml_then_body_child[0]))
+        else_body_ir = IREngine(path_to_xml=None,
+                                path_to_bin=self.path_to_bin,
+                                xml_tree=ElementTree(xml_else_body_child[0]))
+
+        self.graph.graph['hashes'].update(then_body_ir.graph.graph['hashes'])
+        self.graph.graph['hashes'].update(else_body_ir.graph.graph['hashes'])
+
+        # Find port_map section and take an input_port_map & output_port_map
+        xml_then_port_map = list(layer.iterfind('then_port_map'))
+        if not len(xml_then_port_map) == 1:
+            log.warning("If then_body won\'t be compared due to missing then_port_map section!")
```

Comment:
Please, add a operation name to these messages.

Reply:
done